alinka/widget/containers/main_body/content/settings/settings_tabs/team_member_tab/team_member_table_group.py

```diff
@@ -8,6 +7,98 @@
     QVBoxLayout,
     QWidget,
 )
+from sqlalchemy import delete
+
+from alinka.db.connection import db_session
+from alinka.db.models import TeamMember
+from alinka.db.queries import get_team_members, upsert_team_members
+from alinka.schemas import TeamMemberDbSchema
+
+
+class TeamMemberTableModel(QAbstractTableModel):
+    def __init__(self):
+        self.insert_row = None
+        super().__init__()
+
+    def columnCount(self, parent=QModelIndex()):
+        return len(TeamMember.__table__.columns)
+
+    def data(self, index, role=Qt.DisplayRole):
+        """Return the data for the given index."""
+        if not index.isValid():
+            return None
+        if role == Qt.DisplayRole:
+            if self.insert_row is not None and index.row() + 1 == self.rowCount():
+                return self.insert_row.get(TeamMember.__table__.columns.keys()[index.column()], "")
+            return getattr(
+                get_team_members()[index.row()],
+                TeamMember.__table__.columns.keys()[index.column()],
+            )
+        return None
+
+    def headerData(self, section, orientation, role=Qt.DisplayRole):
+        if role == Qt.DisplayRole:
+            if orientation == Qt.Horizontal:
+                return TeamMember.__table__.columns.keys()[section]
+            if orientation == Qt.Vertical:
+                return section + 1
+        return None
+
+    def rowCount(self, parent=QModelIndex()):
+        return len(get_team_members()) + (1 if self.insert_row is not None else 0)
+
+    def flags(self, index):
+        """Set flags for each cell."""
+        if not index.isValid():
+            return Qt.NoItemFlags
+
+        return Qt.ItemIsEnabled | Qt.ItemIsSelectable | Qt.ItemIsEditable | Qt.ItemIsUserCheckable
+
+    def setData(self, index, value, role=Qt.EditRole):
+        if not index.isValid():
+            return False
+        if self.insert_row is not None and index.row() + 1 == self.rowCount():
```

Comment:
WDYT to split it into more meaningful lines?
```suggestion
        new_team_member_added = bool(self.insert_row is not None and index.row() + 1 == self.rowCount())
        if new_team_member_added:
```

Reply:
I was thinking about it and I believe whole validation framework we currently have need a re-write. You will see in new version in this pull request me doing something alonside:
```python
if all(self.insert_row.values()):
    try:
        tm = TeamMemberDbCreateSchema.model_validate(self.insert_row)
    except ValidationError:
        pass
```
which feels terrible (but I don't want want and don't know how to rewrite our validation logic, at least yet).
Until that, I would prefer to not touch this logic.